### PR TITLE
Add default 90 day bucket retention

### PR DESCRIPTION
**Pull request by m0dular**

### CODE REVIEW

File: spec/unit/puppet/provider/influxdb_bucket/influxdb_bucket_spec.rb [hunk 1]
```diff
@@ -166,402 +166,6 @@
         expect(provider.get(context)).to eq should_hash
       end
     end
-
-    # context 'with paginated api response' do
```
Comment (MartyEwings) [inline spec/unit/puppet/provider/influxdb_bucket/influxdb_bucket_spec.rb:170]: What's all this getting removed?

**Review by elainemccloskey — APPROVED** [review]

**Comment by bastelfreak** [comment]
@m0dular is that a breaking change that will change the retention of existing buckets from infinity to 90 days? Or only for new buckets?

**Comment by SimonHoenscheid** [comment]
I would be nice if the retention time would be managable via parameter. We often have use cases where date >3 months are needed

**Comment by m0dular** [comment]
@bastelfreak @SimonHoenscheid Since `influxdb_bucket` is a resource, I don't think a parameter in the module's Puppet code could be used.  You can pass your own `retention_rules` to any buckets that you manage, or use a resource default in your manifests to set it.

**Pull request merged**